### ISSUE TITLE
Refactor fit style as FullWidth container component

### ISSUE THREAD

**Issue by pedrottimark** [FormidableLabs/spectacle#116]
What do you think of a FullWidth component to cause its children to scale to the full width instead of the fit prop for particular components?

Goals:
1. Deduplicate code in Heading and Text components, especially lifecycle methods.
2. Be able to apply this layout in more situations (see use cases below).
3. Hopefully not mess up too badly if someone uses it in a weird way.

The key to correct alignment of multiple child elements seems to be transformOrigin: "left top" instead of "center top" and also to set the width of the inner container after its display becomes block to what its offsetWidth was while its display was inline-block.

``` html
<Heading size={1} fit caps>A ReactJS Presentation Library</Heading>
<FullWidth><Heading size={1} caps>A ReactJS Presentation Library</Heading></FullWidth>
```

``` html
<FullWidth>
  <BlockQuote>
    <Quote>Spectacle<br/>is spectacular!</Quote>
    <Cite>Ken Wheeler</Cite>
  </BlockQuote>
</FullWidth>
```

``` html
<FullWidth>
  <List>
    <ListItem>Who’s on first?</ListItem>
    <ListItem>What’s on second?</ListItem>
    <ListItem>I don’t know.</ListItem>
  </List>
</FullWidth>
```

``` html
<Appear><Heading fit size={4}>Three</Heading></Appear>
<Appear><Heading fit size={3}>Two</Heading></Appear>
<Appear><Heading fit size={2}>One</Heading></Appear>
<Appear><Heading fit size={1}>💥</Heading></Appear>

<FullWidth>
  <Appear><Heading fit size={4}>Three</Heading></Appear>
  <Appear><Heading fit size={3}>Two</Heading></Appear>
  <Appear><Heading fit size={2}>One</Heading></Appear>
  <Appear><Heading fit size={1}>💥</Heading></Appear>
</FullWidth>
```

**Comment by pedrottimark** [comment]
What do you think about including lineHeight in base style props to reduce special cases?

Then Text would have no unique style props and Heading would have only size.

**Comment by pedrottimark** [comment]
For your info, the implementation is ready.

It works for the default deck and some additional cases (including use cases above).
It works well with <br/> explicit line breaks.
It even preserves different explicit alignments for different children!

The main surprise was when FullWidth contains several lines which have a narrow max width, then its scaled height is taller than the slide.

**Comment by kenwheeler** [comment]
How does this work with nested fits?

**Comment by pedrottimark** [comment]
Good question. Do you mean
<FullWidth>...<FullWidth>...</FullWidth>...</FullWidth> or
<FullWidth>...<Text fit>...</Text>...</FullWidth> or
interaction with Layout Fit and Fill?

My thought was to remove the fit prop from Heading and Text but provide a console message (imitating React) if someone uses it.

**Comment by kenwheeler** [comment]
Can I see how this works?

**Comment by pedrottimark** [comment]
Yes, for sure. When to do what is an important part of the learning for me here.

I pushed changes to https://github.com/pedrottimark/spectacle/tree/fit-to-full-width
This is one corner of the design space: replace fit prop with FullWidth container.
It is very likely that FullWidth could be private implementation of fit prop,
or it would even be possible to provide both in public API.

The significant changes are in src/components/full-width.js, heading.js, text.js
I rewrote the existing fit props in index.js and appended the exploratory slides.
